File: 백준/Silver/1149. RGB거리/RGB거리.cc
#include    <iostream>
#include    <algorithm>
using namespace std;

int n;
int dp[1001][3];
int arr[1001][3];

int main()
{
    cin>>n;
    for(int i=1;i<=n;i++)
    {
        cin>>arr[i][0]>>arr[i][1]>>arr[i][2];
        dp[i][0]=min(dp[i-1][1],dp[i-1][2])+arr[i][0];
        dp[i][1]=min(dp[i-1][0],dp[i-1][2])+arr[i][1];
        dp[i][2]=min(dp[i-1][0],dp[i-1][1])+arr[i][2];
    }
    cout<<min(dp[n][0],min(dp[n][1],dp[n][2]))<<endl;
}

File: 백준/Silver/1149. RGB거리/RGB거리.java
//  RGB거리

import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int arr[][] = new int[n][3];
        int dp[][] = new int[n][3];
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            for (int j = 0; j < 3; j++) {
                arr[i][j] = Integer.parseInt(st.nextToken());
                dp[i][j] = Integer.MAX_VALUE;
            }
        }
        // dp 배열에 현재 해당 값으로 색칠할 수 있는 누적값 중 가장 최소의 값을 구하자
        // 0:R / 1:G / 2:B
        dp[0][0] = arr[0][0];
        dp[0][1] = arr[0][1];
        dp[0][2] = arr[0][2];
        for (int i = 1; i < n; i++) {
            dp[i][0] = Math.min(dp[i - 1][1], dp[i - 1][2]) + arr[i][0];
            dp[i][1] = Math.min(dp[i - 1][0], dp[i - 1][2]) + arr[i][1];
            dp[i][2] = Math.min(dp[i - 1][0], dp[i - 1][1]) + arr[i][2];
        }
        int answer = Math.min(dp[n - 1][0], dp[n - 1][1]);
        answer = Math.min(answer, dp[n - 1][2]);
        System.out.println(answer);
    }
}
